packages/api/src/eventListener.ts

```diff
@@ -819,6 +839,12 @@ export default class EventListener extends IndexPriceInterface {
 		newIndexPrice: number,
 	) {
 		const fundingRate = this.fundingRate.get(perpetualId) || 0;
+		// Do not allow 0 funding rate to be sent out
+		if (fundingRate === 0) {
+			console.log(`[updateMarkPrice] funding rate is 0 for ${perpetualId}`);
+			return;
+		}
+
```

Comment:
same as above, should be an error

Reply:
adjusted

Reply:
@b1674927 are you sure it is not possible to have 0 fundingRate for a perpetual? On arbitrum sepolia perp `30004` currently has 0 funding rate:

```json
{
"id": 300004,
"state": "NORMAL",
"baseCurrency": "WIF",
"quoteCurrency": "USD",
"indexPrice": 1,
"collToQuoteIndexPrice": 1.000005,
"markPrice": 1,
"midPrice": 1,
"currentFundingRateBps": 0,
"openInterestBC": 0,
"isMarketClosed": false
}
```

This would basically crash the backend

Reply:
Good observation, as soon as the perp is traded for the first time, the funding rate is non-zero.